webapp/channels/src/utils/performance_telemetry/reporter.ts

```diff
@@ -0,0 +1,277 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import type {Store} from 'redux';
+import {onCLS, onFCP, onINP, onLCP, onTTFB} from 'web-vitals';
+import type {Metric} from 'web-vitals';
+
+import type {Client4} from '@mattermost/client';
+
+import {getConfig} from 'mattermost-redux/selectors/entities/general';
+import {getCurrentUserId} from 'mattermost-redux/selectors/entities/users';
+
+import type {GlobalState} from 'types/store';
+
+import type {PerformanceLongTaskTiming} from './long_task';
+
+type PerformanceReportMeasure = {
+    metric: string;
+    value: number;
+    timestamp: number;
+}
+
+type PerformanceReport = {
+
+    // TODO confirm the version number that we want to start with
+    version: '0';
```

Comment:
I think we can start with `1.0` for now. I'd like to have it some semver semantics so that it would be easier for us to maintain in the future and ensure interoperability within minor changes.

Reply:
I could've swore I fixed that. Thanks for the reminder